op-node/cmd/genesis/cmd.go

```diff
@@ -191,13 +191,13 @@ var Subcommands = cli.Commands{
 			}
 
 			// Build the L2 genesis block
-			l2Genesis, err := genesis.BuildL2Genesis(config, l2Allocs, l1StartBlock.Header())
+			l2Genesis, err := genesis.BuildL2Genesis(config, l2Allocs, genesis.BlockRefFromHeader(l1StartBlock.Header(), nil))
```

Comment:
@mslipper technically, here it's straightforward to derive the correct hash and feed it to the block ref (instead of the blockRef deriving it from the ethclient's Header like it does in here) just via an accessible RpcClient.

But not sure if we want to incorporate that change here in this PR that's why I skipped doing that.

If you think otherwise, feel free to let me know, it's gonna be a quick change.